### DISCUSSION TITLE
Pitch based cues for line length to help navigate large documents

### DISCUSSION THREAD

**Discussion by GPeye** [nvaccess/nvda#14022] [Ideas]
Hello All,

I'm software developer and a sighted person who recently became a new Dad and has plenty of things pulling me away from my desk which started me on a journey of discovery of how it might be possible to still write code while being away from my desk. One line of thinking was how do people who cannot see their screen do it? This led me here to NVDA. I've read around but I'm still a noob so I apologize if this isn't useful or is covered by some addon I haven't found. Perhaps there's some much more obvious way to do this that I have missed.

As a sighted person I struggle with simply knowing where I am in a document if I'm not looking at it and I commend anyone who can. So my idea is around quickly understanding the layout of a document and easily navigating to a specific part of it. This is mostly focused on code documents but I imagine it could have benefits elsewhere. There are a couple ways to implement this but essentially it would revolve around playing a variable pitch based on the line while you skim down the page, before NVDA reads the line. 

For example that pitch could be based on the line length. 0 characters would be a low pitch and a really long line of characters would be a high pitch. This would let you just hold the up or down arrow to move the cursor through a document and hear an array of pitches that outlines the document, helping you make a mental map. Obviously this becomes less necessary the more familiar you become with a document and it's layout but would help in that initial orientation or if you revisit a doc you haven't been in for some time.

A different implementation could be the pitch would depend on the current number of tabs at the beginning of a line. This would give you probably a more useful idea of the structure of a document of code and the various scopes within it. I can imagine opening a doc of code, holding the down arrow and listening for "Yep there's the imports, there's the declaration of the class, it's properties block, first function likely a constructor, function 1, function 2, end of class scope, I probably need to edit one of those last 2 functions so now I'll hold the up arrow until I hear that I'm back to one them and pause so that NVDA can read the line and confirm I'm where I want to be" et cetera.

So I'm curious what anyone else thinks. Does this idea have any merit? Or does this only make sense in my head.

**Comment by trypsynth** [comment]
Hi there,
NVDA is able to beep for indentation. Open the NVDA menu, preferences, settings, document formatting, Line indentation reporting. I have mine set to tones. There's also an add-on called indent-nav that lets you quickly navigate through blocks with the same indentation level, like a tree. https://github.com/mltony/nvda-indent-nav/. And as regards the document positions thing, there's an add-on called percentage-checker that lets you press NVDA+shift+p and have the percentage spoken. It works in text fields and list boxes. https://github.com/lukaszgo1/percentageChecker.
Hope this helped!
